src/main/java/seedu/address/logic/commands/Filter.java

```diff
@@ -2,34 +2,36 @@
 
 import static java.util.Objects.requireNonNull;
 
+import java.util.function.Predicate;
+
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
 import seedu.address.model.Model;
-import seedu.address.model.person.PersonHasTagPredicate;
+import seedu.address.model.person.Person;
 
 /**
- * Sorts and lists all persons in address book who are tagged by any of the argument keywords.
- * Keyword matching is case-sensitive.
+ * A common superclass for all filter commands that have the same logic, but filter using
+ * different predicates.
  */
-public class FilterCommand extends Command {
+public abstract class Filter extends Command {
     public static final String COMMAND_WORD = "filter";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters for all persons whose tags include any of "
-            + "the specified keywords and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " friend teacher student";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters the contact list according to one of the "
+            + "two possible fields: days and tags and displays them as a list with index numbers.\n"
+            + "Parameters: filter [type] [keywords]...\n"
+            + "Example 1: " + COMMAND_WORD + "tag student"
```

Comment:
I think from a user perspective it might be more familiar for them to use filter [type] [keywords] since this is the same pattern as your sort command. Will try to find a way to implement this without the magic number!